From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>
Date: Tue, 9 Nov 2077 00:00:00 +0800
Subject: [PATCH] Short-circuit isOnFire checks

Reorder checks to short-circuit earlier and skip client-side checks.

diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index b25976d56c9b2db983a0fe97f67db37596859658..3f9393cb9cf7630023c7a069ef08de91404286f9 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -3857,8 +3857,10 @@ public abstract class Entity
     }
 
     public boolean isOnFire() {
-        boolean isClientSide = this.level() != null && this.level().isClientSide();
-        return !this.fireImmune() && (this.remainingFireTicks > 0 || isClientSide && this.getSharedFlag(FLAG_ONFIRE));
+        // Leaf start - Short-circuit isOnFire checks
+        //boolean isClientSide = this.level() != null && this.level().isClientSide();
+        return (this.remainingFireTicks > 0 /*|| isClientSide && this.getSharedFlag(FLAG_ONFIRE)*/) && !this.fireImmune();
+        // Leaf end - Short-circuit isOnFire checks
     }
 
     public boolean isPassenger() {
